components/board/customlinux/setup.go

```diff
@@ -0,0 +1,83 @@
+// Package customlinux implements a board running linux
+package customlinux
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/pkg/errors"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/board/genericlinux"
+)
+
+// GenericLinuxPin describes a gpio pin on a linux board.
+type GenericLinuxPin struct {
+	Name            string `json:"name"`
+	Ngpio           int    `json:"ngpio"`
+	RelativeID      int    `json:"relative_id"`
+	PWMChipSysFSDir string `json:"pwm_chip_dir,omitempty"`
+	PWMID           int    `json:"pwm_id,omitempty"`
```

Comment:
So many capital letters in a row! I have a very mild preference for `PwmChipSysfsDir` (lowercase WM and FS), and `PwmId` (lowercase WM and D).